core/src/main/java/org/apache/iceberg/SnapshotSummary.java

```diff
@@ -148,7 +148,7 @@ public void set(String property, String value) {
     }
 
     private void updatePartitions(PartitionSpec spec, ContentFile<?> file, boolean isAddition) {
-      if (trustPartitionMetrics) {
+      if (spec.isPartitioned() && trustPartitionMetrics) {
```

Comment:
I don't think we need to do anything else. This PR is is only so that the snapshot summary doesn't produce a hashmap entry where the key is `partitions.`